compiler-core/src/io.rs

```diff
@@ -278,11 +278,25 @@ pub trait FileSystemReader {
 /// Symlinks are followed.
 pub fn gleam_source_files(io: &impl FileSystemReader, dir: &Utf8Path) -> Vec<Utf8PathBuf> {
     tracing::trace!("gleam_source_files {:?}", dir);
+    files_with_extension(io, dir, "gleam")
+}
+
+/// Iterates over Gleam cache files (`.cache`) in a certain directory.
+/// Symlinks are followed.
+pub fn gleam_cache_files(io: &impl FileSystemReader, dir: &Utf8Path) -> Vec<Utf8PathBuf> {
+    tracing::trace!("gleam_cache_files {:?}", dir);
+    files_with_extension(io, dir, "cache")
+}
 
+fn files_with_extension(
+    io: &impl FileSystemReader,
+    dir: &Utf8Path,
+    extension: &str,
+) -> Vec<Utf8PathBuf> {
```

Comment:
We could also make this
```rs
fn files_with_extension<'a>(
    io: &'a impl FileSystemReader,
    dir: &'a Utf8Path,
    extension: &'a str,
) -> impl Iterator<Item = Utf8PathBuf> + 'a {
```

(with similar changes to the wrapping functions, if we want to avoid the `collect()`

Reply:
I'm happy to merge this, but make a PR if you want to optimise this 🙏